p2p/netaddress.go

```diff
@@ -41,10 +42,10 @@ func IDAddressString(id ID, protocolHostPort string) string {
 // panic. Panics if ID is invalid.
 // TODO: socks proxies?
 func NewNetAddress(id ID, addr net.Addr) *NetAddress {
-	tcpAddr, ok := addr.(*net.TCPAddr)
+	tcpAddr, ok := addr.(*net.UDPAddr)
 	if !ok {
 		if flag.Lookup("test.v") == nil { // normal run
-			panic(fmt.Sprintf("Only TCPAddrs are supported. Got: %v", addr))
+			panic(fmt.Sprintf("Only UDPAddrs are supported. Got: %v", addr))
```

Comment:
If you want to add support for a novel transport-level protocol like KCP (which also switches the network-layer protocol from TCP to UDP) you can't just redefine a core function like NewNetAddress to use your alternative by default, you have to make it opt-in.